### PR TITLE
Format src/Compiler/SyntaxTree

### PR DESCRIPTION
**Pull request by dsyme**

### CODE REVIEW

File: src/Compiler/SyntaxTree/LexFilter.fs [hunk 1]
```diff
@@ -697,62 +840,75 @@ type LexFilterImpl (
...
-        let rec check s = 
-               match s with 
-               | CtxtParen(LBRACE _, _) :: _ -> true
-               | (CtxtSeqBlock _ | CtxtDo _ | CtxtFor _) :: rest -> check rest
-               | _ -> false
-        match stack with 
+        let rec check s =
+            match s with
+            | CtxtParen (LBRACE _, _) :: _ -> true
+            | (CtxtSeqBlock _
```
Comment (dsyme) [inline src/Compiler/SyntaxTree/LexFilter.fs:871]: These disjunctive patterns look odd

Comment (dsyme) [reply src/Compiler/SyntaxTree/LexFilter.fs]: I removed LexFilter.fs from the formatting.  It's just got a huge nubmer of disjunctive patterns. which cause problems

**Comment by dsyme** [comment]
This is ready for review

**Review by KevinRansom — APPROVED** [review]

File: src/Compiler/SyntaxTree/PrettyNaming.fs [hunk 2]
```diff
@@ -474,33 +520,35 @@ let DecompileOpName opName =
             opName
 
 let DoesIdentifierNeedBackticks (name: string) : bool =
-    not (IsUnencodedOpName name) && 
-    not (IsIdentifierName name) && 
-    not (IsActivePatternName name)
+    not (IsUnencodedOpName name)
+    && not (IsIdentifierName name)
+    && not (IsActivePatternName name)
```
Comment (auduchinok) [inline src/Compiler/SyntaxTree/PrettyNaming.fs:525]: @dsyme it looks like there are many places where operators got recently moved to the next line and the most of the compiler codebase used to have it on the previous line. What do you think about making it a preferred default in the style guide to have it on the previous line, as it used to be?

File: src/Compiler/SyntaxTree/LexHelpers.fs [hunk 3]
```diff
@@ -68,33 +75,41 @@ type LongUnicodeLexResult =
...
-    lexbuf.EndPos <- Position.FirstLine (FileIndex.fileIndexOfFile fileName)
+        raise (
+            WrappedError(
+                e,
+                (try
+                    lexbuf.LexemeRange
+                 with _ ->
+                     range0)
+            )
+        )
```
Comment (auduchinok) [inline src/Compiler/SyntaxTree/LexHelpers.fs:106]: @dsyme @nojaf Could we allow Fantomas to keep such oneliners? Is there a style guide rule for this?

Comment (dsyme) [reply src/Compiler/SyntaxTree/LexHelpers.fs:106]: This code is badly written, it should use a let binding

I'm OK with try/with always being multiline. There should be fewer of these in the code base anyway

File: src/Compiler/SyntaxTree/PrettyNaming.fs (hunk 2 above)
Comment (dsyme) [reply src/Compiler/SyntaxTree/PrettyNaming.fs:525]: If fantomas has this option I'd consider using it. However I cant see such an option here? https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md

Comment (auduchinok) [reply src/Compiler/SyntaxTree/PrettyNaming.fs:525]: @nojaf What do you think about adding such an option?

Comment (nojaf) [reply src/Compiler/SyntaxTree/PrettyNaming.fs:525]: https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#what-are-we-not-looking-for

This has been going on for some weeks now, so allow me to spell it out one last time:
The style is determined by the guide, which is discussed over at [fsharp/fs-langdesign](https://github.com/fsharp/fslang-design#style-guide).
Please don't ask for settings but convince your peers of your idea so we can have better defaults.

Comment (auduchinok) [reply src/Compiler/SyntaxTree/PrettyNaming.fs:525]: Yes, I've started with the style guide question.

**Pull request merged**